sql/moz-fx-data-shared-prod/search_derived/search_clients_last_seen_v2/query.sql

```diff
@@ -0,0 +1,158 @@
+WITH _derived_search_cols AS (
+  SELECT
+    COALESCE(udf.normalize_search_engine(engine), "Other") AS short_engine,
+    COALESCE(organic, 0) + COALESCE(sap, 0) + COALESCE(unknown, 0) + COALESCE(
+      tagged_sap,
+      0
+    ) + COALESCE(tagged_follow_on, 0) AS total_searches,
+    COALESCE(tagged_sap, 0) + COALESCE(tagged_follow_on, 0) AS tagged_searches,
+    COALESCE(ad_click, 0) AS ad_click,
+    COALESCE(search_with_ads, 0) AS search_with_ads,
+    scd.* EXCEPT (ad_click, search_with_ads),
+    first_seen_date,
+    cfs.country AS first_reported_country,
+    DATE_DIFF(@submission_date, first_seen_date, DAY) AS days_since_first_seen
+  FROM
+    `moz-fx-data-shared-prod.search_derived.search_clients_daily_v8` scd
+  LEFT JOIN
+    `moz-fx-data-shared-prod.telemetry.clients_first_seen` cfs
+    USING (client_id)
+  WHERE
+    scd.submission_date = @submission_date
+),
+_derived_engine_searches AS (
+    -- From the clients search info, make a struct
+    -- that we will use for aggregation later
+  SELECT
+    STRUCT(
+      short_engine AS key,
+      STRUCT(total_searches, tagged_searches, ad_click, search_with_ads) AS value
+    ) AS engine_searches,
+    * EXCEPT (short_engine)
+  FROM
+    _derived_search_cols
+),
+_grouped AS (
+    -- Here we get a single row per-client, containing
+    -- info from each engine, as well as overall info
+  SELECT
+      -- Grouping columns
+    client_id,
+    sample_id,
+      -- Dimensional data
+    udf.mode_last(ARRAY_AGG(country)) AS country,
+    udf.mode_last(ARRAY_AGG(app_version)) AS app_version,
+    udf.mode_last(ARRAY_AGG(distribution_id)) AS distribution_id,
+    udf.mode_last(ARRAY_AGG(locale)) AS locale,
+    udf.mode_last(ARRAY_AGG(search_cohort)) AS search_cohort,
+    udf.mode_last(ARRAY_AGG(addon_version)) AS addon_version,
+    udf.mode_last(ARRAY_AGG(os)) AS os,
+    udf.mode_last(ARRAY_AGG(channel)) AS channel,
+    udf.mode_last(ARRAY_AGG(first_seen_date)) AS first_seen_date,
+    udf.mode_last(ARRAY_AGG(default_search_engine)) AS default_search_engine,
+    udf.mode_last(
+      ARRAY_AGG(default_search_engine_data_load_path)
+    ) AS default_search_engine_data_load_path,
+    udf.mode_last(
+      ARRAY_AGG(default_search_engine_data_submission_url)
+    ) AS default_search_engine_data_submission_url,
+    udf.mode_last(ARRAY_AGG(days_since_first_seen)) AS days_since_first_seen,
```

Comment:
something is going wrong here or on line 14

Reply:
we figured out the problem, its that on days the client doesn't have any entries in search_clients_daily_v8, we carry forward their old value. what we need to do is calculate it every day, even if they didn't come back.

Reply:
we decided to just remove this column since it can be easily calculated at use time